checker/src/main/java/org/checkerframework/checker/determinism/DeterminismVisitor.java

```diff
@@ -525,6 +526,59 @@ public Void visitMethod(MethodTree node, Void p) {
         return super.visitMethod(node, p);
     }
 
+    /**
+     * If the declaration of {@code node} is annotated with {@code @RequiresDetToString}, checks
+     * that the declared type of every {@code Det} argument of {@code node} corresponding to a
+     * parameter of type {@code Object} overrides {@code toString} returning a {@code @Det String}
+     * and reports an error otherwise.
+     */
+    @Override
+    public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
+        ExecutableElement methodElement = TreeUtils.elementFromUse(node);
+        AnnotationMirror declAnnotation =
+                atypeFactory.getDeclAnnotation(methodElement, RequiresDetToString.class);
+        if (declAnnotation != null) {
+            List<? extends VariableElement> params = methodElement.getParameters();
+            List<? extends ExpressionTree> args = node.getArguments();
+
+            for (int index = 0; index < args.size(); index++) {
+                ExpressionTree arg = args.get(index);
+                VariableElement param = params.get(index);
+
+                boolean isParamObjectArray = false;
+                TypeMirror paramType = param.asType();
+                if (paramType.getKind() == TypeKind.ARRAY) {
+                    TypeMirror compType = ((ArrayType) paramType).getComponentType();
+                    isParamObjectArray = TypesUtils.isObject(compType);
+                }
+                if (TypesUtils.isObject(paramType) || isParamObjectArray) {
+                    AnnotatedTypeMirror argType = atypeFactory.getAnnotatedType(arg);
+                    if (argType.hasAnnotation(atypeFactory.DET)) {
+                        Pair<AnnotatedDeclaredType, ExecutableElement> overriddenMethod =
+                                AnnotatedTypes.getOverriddenMethod(
+                                        argType, stringToString, atypeFactory.getProcessingEnv());
+                        if (!atypeFactory
```

Comment:
The indentation here is pretty deep which makes this hard to read.  On way to fix this is to invert some of the if statements and return/continue.  For example, This first if statement in this method could be:
```
 if (declAnnotation == null) {
    return super.visitMethodInvocation(node, p);
}
```